pytket/docs/circuit_class.rst

```diff
@@ -20,7 +20,322 @@ following keyword arguments are supported:
 ``condition``, or using the pair ``condition_bits`` and ``condition_value`` to
 condition on a specified set of bit values.)
 
-.. currentmodule:: pytket._tket.circuit.Circuit
-.. autoclass:: pytket._tket.circuit.Circuit
-   :special-members:
-   :members:
+.. currentmodule:: pytket.circuit.Circuit
+.. autoclass:: pytket.circuit.Circuit
+   :special-members: __init__, __eq__, __iter__, __mul__, __repr__, __rshift__, __str__, __getstate__, __setstate__, __hash__
```

Comment:
Good point about the `__init__` method. Will try to fix that.

I'm not sure what you mean by 
```
It should also mainly be examples to show what each way of initialising means
```

Could you reword please?

Reply:
`__init__`, `__eq__` etc all now appear at the top. I've removed the `:special-members:` and used `:automethod:` following this answer -> https://stackoverflow.com/questions/42214656/sphinx-naming-special-members

Reply:
![image](https://github.com/CQCL/tket/assets/12997250/6f48cfd8-cebb-4d0d-ba8b-5215bddec9ca)

This is quite hard to understand, and constructing a circuit is one of the most important API entry points.
It would be better to go through a series of examples, something like:


```python
# initialise an empty circuit
c = Circuit()
# Initialise a circuit with 3 qubit
c = Circuit(3)
# initialise a circuit with 3 qubits and 3 bits
c = Circuit(3, 3)
# Optionally give a circuit a name
c = Circuit(3, name="mycircuit")

Reply:
Added some snippets on initialisation

Reply:
where? I can't find it

Reply:
Reverted the change as there was a weird issue I was having with the `pytket/binders/circuit/Circuit/main.cpp` file that was causing the CI checks to fail. Didn't figure out why.

Maybe I can add these in in a separate PR.